src/types/transaction.cairo

```diff
@@ -132,13 +132,13 @@ pub impl TransactionImpl of TransactionTrait {
     /// NOTE: marker, flag, and witness fields in segwit transactions are not included
     /// this means txid computation is the same for legacy and segwit tx
     fn txid(self: @Transaction) -> Hash {
-        double_sha256_byte_array(@encode_transaction(self, false))
+        double_sha256_byte_array(@EncodeTrait::encode(*self, false))
```

Comment:
you can do self.encode

Reply:
Instead of desnapping the transaction, wouldn't it be better to transform the implementation into EncodeTrait<@Transaction>?

Reply:
You can implement `EncodeSnap<T, +Encode<T>> of Encode<@T>`